msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/DeviceCodeFlowAPITest.java

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.e2e.tests.mocked;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.client.AuthenticationResult;
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.e2e.shadows.ShadowDeviceCodeFlowCommandAuthError;
+import com.microsoft.identity.client.e2e.shadows.ShadowDeviceCodeFlowCommandSuccessful;
+import com.microsoft.identity.client.e2e.shadows.ShadowDeviceCodeFlowCommandTokenError;
+import com.microsoft.identity.client.e2e.shadows.ShadowHttpRequestForMockedTest;
+import com.microsoft.identity.client.e2e.shadows.ShadowMsalUtils;
+import com.microsoft.identity.client.e2e.tests.PublicClientApplicationAbstractTest;
+import com.microsoft.identity.client.e2e.utils.RoboTestUtils;
+import com.microsoft.identity.client.exception.MsalException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static com.microsoft.identity.internal.testutils.TestConstants.Configurations.SINGLE_ACCOUNT_DCF_TEST_CONFIG_FILE_PATH;
+
+/**
+ * Testing class for the device code flow protocol. Currently only supporting testing for the API-side
+ * of the protocol. Will be extended to test individual aspects of the flow.
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(shadows = {ShadowMsalUtils.class, ShadowHttpRequestForMockedTest.class})
+public class DeviceCodeFlowAPITest extends PublicClientApplicationAbstractTest {
+
+    private static Boolean uCode;
```

Comment:
Can this be a primitive?

Reply:
Yes, it can be primitive. No, t doesn't need to be static.